txstratum/manager.py

```diff
@@ -264,7 +273,10 @@ def _job_timeout_if_possible(self, job: TxJob) -> None:
         self.log.info('TxJob timeout', job=job.to_dict())
         self.txs_timeout += 1
         job.status = JobStatus.TIMEOUT
-        self.tx_queue.remove(job)
+        try:
+            self.tx_queue.remove(job)
+        except ValueError:
+            self.log.warning('TxJob timeout but not in queue', job=job)
```

Comment:
IMO the correct implementation wouldn't necessarily _remove_ the job when it is submitted. If this is the case, then jobs would _only_ be removed by timeout, and then it would really make sense to expect or at least log an error when the jobs is not (because it wouldn't be possible for it to have been removed in other way).

Reply:
But if we don't remove a job from the queue after it is submitted, won't it keep being sent to miners until the timeout is reached? Or maybe I'm not getting the whole picture you have in mind.

Reply:
I think I get it now, mostly because it's a `TxJob` instead of a `MinerTxJob`.